control-plane/agents/src/bin/core/controller/reconciler/nexus/mod.rs

```diff
@@ -130,61 +135,126 @@ async fn nexus_reconciler(
+        faulted_children_remover(nexus_spec, child, context).await?
+    } else if get_child_replica(nexus_spec.as_ref(), child, context)
         .await
+        .is_ok()
+    {
+        info!(%child.uri, "Child's Replica is Online within the partial rebuild window");
+        if let Err(error) = initialize_partial_rebuild(nexus_spec, &child.uri, context).await {
+            info!(%child.uri, "Initializing Full rebuild as online child attempt failed for child, failed with : {:?}", error);
+            faulted_children_remover(nexus_spec, child, context).await?
+        }
+    } else {
+        info!(%child.uri, "Child's Replica is not online yet");
     }
+    Ok(())
+}
 
-    PollResult::Ok(PollerState::Idle)
+/// Removes child from nexus children list to initiate Full rebuild of child.
+#[tracing::instrument(skip(nexus, context), level = "trace", fields(nexus.uuid = %nexus.uuid(), request.reconcile = true))]
+pub(super) async fn faulted_children_remover(
+    nexus: &mut OperationGuardArc<NexusSpec>,
+    child: &Child,
+    context: &PollContext,
+) -> Result<(), SvcError> {
+    let nexus_uuid = nexus.uuid();
+    let nexus_state = context.registry().nexus(nexus_uuid).await?;
+    let nexus_spec_clone = nexus.lock().clone();
+    nexus_spec_clone.warn_span(|| {
+        tracing::warn!(%child.uri, %child.state, %child.state_reason, "Attempting to remove faulted child")
+    });
+    nexus
+        .remove_child_by_uri(context.registry(), &nexus_state, &child.uri, true)
+        .await?;
+
+    nexus_spec_clone.warn_span(|| {
+        tracing::warn!(%child.uri, %child.state, %child.state_reason, "Successfully removed faulted child")
+    });
+    Ok(())
+}
+
+/// Returns true if time elapsed from child fault time is greater or equal to wait time duration.
+fn is_time_elapsed(
+    fault_time: Option<SystemTime>,
+    wait_duration: Duration,
+    uri: &ChildUri,
+) -> bool {
+    if let Some(fault_time) = fault_time {
+        if let Ok(elapsed) = fault_time.elapsed() {
+            info!(
+                child.uri = %uri,
+                "waited {:?} for child to comeback. Wait time is {:?}",
+                elapsed, wait_duration
+            );
+            return elapsed >= wait_duration;
+        }
+    }
+    false
+}
+
+/// If we can get replica for a child from registry. It means that child node is online, Pool is
+/// imported containing the child/replica.
+async fn get_child_replica(
+    nexus: &NexusSpec,
+    child: &Child,
+    context: &PollContext,
+) -> Result<Replica, SvcError> {
+    let replica_uri = nexus.replica_uri(&child.uri).ok_or(SvcError::NotFound {
+        kind: ResourceKind::Replica,
+        id: "Child replica not found".to_string(),
+    })?;
+    context.registry().replica(replica_uri.uuid()).await
+}
+
+/// Tries to mark child as Online. Returns error on failing to do so.
+pub(super) async fn initialize_partial_rebuild(
```

Comment:
Suggest a more representative name such as `online_nexus_child` since there is no special partial rebuild call.